code/__HELPERS/spatial_info.dm

```diff
@@ -218,9 +218,15 @@
 /// Modifies input list.
 /proc/mob_only_listeners(list/atom/movable/hearers)
 	RETURN_TYPE(/list/mob)
-	for(var/hearer_index in 1 to hearers.len)
+
+	var/hearers_length
+	if(isnull(hearers) || !(hearers_length = hearers.len))
```

Comment:
Are we setting hearers_length to hearers.len inside a conditional statement for real? Guessing from how the compiler is not throwing a warning about it, I suppose what is happening here is that we're actually checking that hearers_length isn't 0 after the `=` operation, but the syntax honestly got me ngl.

I mean, maybe you're trying to micro-optimize it since it's such a warm proc, but is it better than something like:
```
	var/hearers_length = length(hearers)
	if(!hearers_length)
```
?

Reply:
nvm